packages/build/src/plugins/child/main.js

```diff
@@ -46,6 +46,20 @@ const handleEvent = async function ({
   }
 }
 
-const EVENTS = { load, run }
+const EVENTS = {
+  load,
+  run,
+  // async shutdown hook to stop tracing reliably
+  shutdown: async () => {
+    try {
+      const { stopTracing } = await import('@netlify/opentelemetry-sdk-setup')
+      await stopTracing()
+    } catch {
+      // noop as the opentelemetry-sdk-setup is an optional dependency
+      // and might not be present in the CLI
+    }
+    return { context: {} }
+  },
```

Comment:
Will this work with module preloading given we need `sdk` to be defined here?
- https://github.com/netlify/build/blob/1cee56d47237e07daa078e8149c411aa346bc8e1/packages/opentelemetry-sdk-setup/src/sdk-setup.ts#L80-L81

I wonder if instead it would provide a cleaner solution if we handle the `SIGINT` and `SIGTERM` signals in:
- https://github.com/netlify/build/blob/1cee56d47237e07daa078e8149c411aa346bc8e1/packages/opentelemetry-sdk-setup/src/bin.ts#L64-L66

E.g.
```js
process.once('SIGTERM', async () => {
  await stopTracing()
  process.exit()
})
```

This way we wouldn't need to export the shutdown method and leak the SDK interface around.

Reply:
Yes this works! Sadly the SIGTERM approach did not work we've tried that! sigterm was not called for the child process (the way how we disconnect first and then kill it)

That was the only solution to get it working. And yea it works if no SDK is provided (therefore the dynamic import that is try catched)

Reply:
> Sadly the SIGTERM approach did not work we've tried that! sigterm was not called for the child process (the way how we disconnect first and then kill it)

I wouldn't expect the disconnect to interfere with us sending a signal to the process 🤔 given we're doing `.kill()` afterwards the child process should be getting a SIGTERM and should be able to handle it, unless there's something fishy about the module preloading and the process event listeners somehow. I'll take your word for it though, but I would like us to have a follow up working item to investigate and clean this up if possible since this defeats the whole purpose of scoping things into the SDK if we suddenly need to optionally required it for shutdown every time we use it.